File: backend/WebApi/Lunchorder.Domain/Entities/DocumentDb/MenuVendor.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Lunchorder.Domain.Entities.DocumentDb
{
    /// <summary>
    /// Details of the menu vendor
    /// </summary>
    public class MenuVendor
    {
        /// <summary>
        /// The id of the menu vendor
        /// </summary>
        [JsonProperty("id")]
        public Guid Id { get; set; }

        /// <summary>
        /// The name of the vendor
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The address of the menu vendor
        /// </summary>
        public MenuVendorAddress Address { get; set; }

        /// <summary>
        /// The website of the menu vendor
        /// </summary>
        public string Website { get; set; }

        /// <summary>
        /// The ultimate time limit that an order should be submitted to the vendor
        /// </summary>
        public DateTime SubmitOrderTime { get; set; }

        /// <summary>
        /// Logo of the vendor
        /// </summary>
        public string Logo { get; set; }

        /// <summary>
        /// Enumerable of closing dates of this vendor
        /// </summary>
        public IEnumerable<MenuVendorClosingDateRange> ClosingDateRanges { get; set; }
    }
}

File: frontend/src/app/domain/dto/menuVendor.ts
import { MenuVendorAddress } from './menuVendorAddress'
import { MenuVendorClosingDateRange } from './menuVendorClosingDateRange';

export class MenuVendor implements app.domain.dto.IMenuVendor, Serializable<MenuVendor> {
    id: string;
    name: string;
    address: MenuVendorAddress;
    website: string;
    submitOrderTime: string;
    logo: string;
    closingDateRanges: MenuVendorClosingDateRange[];

    isClosingDate(): boolean {
        var today = new Date();
        for (var closingDateRange of this.closingDateRanges) {
            var fromDate = new Date(closingDateRange.from);
            var untilDate = new Date(closingDateRange.until);

            if (today >= fromDate && today <= untilDate){
                return true;
            }
        }
        return false;
    }

    deserialize(input: any): MenuVendor {
        if (!input) { return; }
        this.id = input.id;
        this.name = input.name;
        this.address = new MenuVendorAddress().deserialize(input.address);
        this.website = input.website;
        this.submitOrderTime = input.submitOrderTime;
        this.logo = input.logo;

        this.closingDateRanges = new Array<MenuVendorClosingDateRange>();
        if (input.closingDateRanges) {
            for (var closingDateRange of input.closingDateRanges) {
                this.closingDateRanges.push(closingDateRange);
            }
        }

        return this;
    }
}
